includes/class-amp-theme-support.php

```diff
@@ -417,15 +419,26 @@ public static function finish_init() {
 
 		self::ensure_proper_amp_location();
 
-		$theme_support = self::get_theme_support_args();
+		$is_reader_mode = self::READER_MODE_SLUG === self::get_support_mode();
+		$theme_support  = self::get_theme_support_args();
 		if ( ! empty( $theme_support['template_dir'] ) ) {
 			self::add_amp_template_filters();
+		} elseif ( $is_reader_mode ) {
+			add_action(
```

Comment:
```suggestion
			add_filter(
```

Reply:
If you turn the action into a filter, you'll need to actually use the return value as well.

Reply:
I don't understand. The filter callback is returning a value:

```php
return __DIR__ . '/templates/reader-template-loader.php';
```

It's not applying filters here.

Reply:
Nevermind, I'm dumb.